diff -crB gcc-4.7.1/config.sub gcc-4.7.1-seaos/config.sub
*** gcc-4.7.1/config.sub	Tue Nov  1 17:56:32 2011
--- gcc-4.7.1-seaos/config.sub	Fri Apr  5 18:38:35 2013
***************
*** 1333,1339 ****
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
  	      | -sym* | -kopensolaris* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
--- 1333,1339 ----
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
  	      | -sym* | -kopensolaris* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* | -seaos* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
Only in gcc-4.7.1-seaos/gcc/config: seaos.h
diff -crB gcc-4.7.1/gcc/config.gcc gcc-4.7.1-seaos/gcc/config.gcc
*** gcc-4.7.1/gcc/config.gcc	Mon Jun  4 02:51:00 2012
--- gcc-4.7.1-seaos/gcc/config.gcc	Fri Apr  5 18:38:36 2013
***************
*** 643,648 ****
--- 643,654 ----
        ;;
    esac
    ;;
+ *-*-seaos*)
+   extra_parts="crtbegin.o crtend.o"
+   gas=yes
+   gnu_ld=yes
+   default_use_cxa_atexit=yes
+   ;;
  *-*-netbsd*)
    tmake_file="t-slibgcc"
    gas=yes
***************
*** 1225,1230 ****
--- 1231,1241 ----
  	gas=yes
  	gnu_ld=yes
  	;;
+ i[3-7]86-*-seaos*)
+ 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h seaos.h"
+ 	tmake_file="${tmake_file} i386/t-i386elf t-svr4 i386/t-crtstuff"
+ 	use_fixproto=yes
+ 	;;
  i[34567]86-*-linux* | i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-knetbsd*-gnu | i[34567]86-*-gnu* | i[34567]86-*-kopensolaris*-gnu)
  			# Intel 80386's running GNU/*
  			# with ELF format using glibc 2
diff -crB gcc-4.7.1/gcc/doc/cppopts.texi gcc-4.7.1-seaos/gcc/doc/cppopts.texi
*** gcc-4.7.1/gcc/doc/cppopts.texi	Tue Dec 20 12:44:13 2011
--- gcc-4.7.1-seaos/gcc/doc/cppopts.texi	Fri Apr  5 18:38:37 2013
***************
*** 803,809 ****
  Enable special code to work around file systems which only permit very
  short file names, such as MS-DOS@.
  
! @itemx --help
  @itemx --target-help
  @opindex help
  @opindex target-help
--- 803,809 ----
  Enable special code to work around file systems which only permit very
  short file names, such as MS-DOS@.
  
! @item --help
  @itemx --target-help
  @opindex help
  @opindex target-help
diff -crB gcc-4.7.1/gcc/doc/gcc.texi gcc-4.7.1-seaos/gcc/doc/gcc.texi
*** gcc-4.7.1/gcc/doc/gcc.texi	Wed Jun  9 16:46:33 2010
--- gcc-4.7.1-seaos/gcc/doc/gcc.texi	Fri Apr  5 18:38:37 2013
***************
*** 86,94 ****
  @item GNU Press
  @tab Website: www.gnupress.org
  @item a division of the
! @tab General: @tex press@@gnu.org @end tex
  @item Free Software Foundation
! @tab Orders:  @tex sales@@gnu.org @end tex
  @item 51 Franklin Street, Fifth Floor
  @tab Tel 617-542-5942
  @item Boston, MA 02110-1301 USA
--- 86,98 ----
  @item GNU Press
  @tab Website: www.gnupress.org
  @item a division of the
! @tab General: 
! @tex press@@gnu.org 
! @end tex
  @item Free Software Foundation
! @tab Orders:  
! @tex sales@@gnu.org 
! @end tex
  @item 51 Franklin Street, Fifth Floor
  @tab Tel 617-542-5942
  @item Boston, MA 02110-1301 USA
diff -crB gcc-4.7.1/gcc/doc/generic.texi gcc-4.7.1-seaos/gcc/doc/generic.texi
*** gcc-4.7.1/gcc/doc/generic.texi	Fri Dec 23 14:07:16 2011
--- gcc-4.7.1-seaos/gcc/doc/generic.texi	Fri Apr  5 18:38:37 2013
***************
*** 1415,1427 ****
  not matter.  The type of the operands and that of the result are
  always of @code{BOOLEAN_TYPE} or @code{INTEGER_TYPE}.
  
! @itemx POINTER_PLUS_EXPR
  This node represents pointer arithmetic.  The first operand is always
  a pointer/reference type.  The second operand is always an unsigned
  integer type compatible with sizetype.  This is the only binary
  arithmetic operand that can operate on pointer types.
  
! @itemx PLUS_EXPR
  @itemx MINUS_EXPR
  @itemx MULT_EXPR
  These nodes represent various binary arithmetic operations.
--- 1415,1427 ----
  not matter.  The type of the operands and that of the result are
  always of @code{BOOLEAN_TYPE} or @code{INTEGER_TYPE}.
  
! @item POINTER_PLUS_EXPR
  This node represents pointer arithmetic.  The first operand is always
  a pointer/reference type.  The second operand is always an unsigned
  integer type compatible with sizetype.  This is the only binary
  arithmetic operand that can operate on pointer types.
  
! @item PLUS_EXPR
  @itemx MINUS_EXPR
  @itemx MULT_EXPR
  These nodes represent various binary arithmetic operations.
diff -crB gcc-4.7.1/gcc/doc/invoke.texi gcc-4.7.1-seaos/gcc/doc/invoke.texi
*** gcc-4.7.1/gcc/doc/invoke.texi	Mon Jun  4 13:30:59 2012
--- gcc-4.7.1-seaos/gcc/doc/invoke.texi	Fri Apr  5 18:38:37 2013
***************
*** 5178,5184 ****
  e.g. With -fdbg-cnt=dce:10,tail_call:0
  dbg_cnt(dce) will return true only for first 10 invocations
  
! @itemx -fenable-@var{kind}-@var{pass}
  @itemx -fdisable-@var{kind}-@var{pass}=@var{range-list}
  @opindex fdisable-
  @opindex fenable-
--- 5178,5184 ----
  e.g. With -fdbg-cnt=dce:10,tail_call:0
  dbg_cnt(dce) will return true only for first 10 invocations
  
! @item -fenable-@var{kind}-@var{pass}
  @itemx -fdisable-@var{kind}-@var{pass}=@var{range-list}
  @opindex fdisable-
  @opindex fenable-
***************
*** 5326,5336 ****
  @option{-fdump-rtl-ce3} enable dumping after the three
  if conversion passes.
  
! @itemx -fdump-rtl-cprop_hardreg
  @opindex fdump-rtl-cprop_hardreg
  Dump after hard register copy propagation.
  
! @itemx -fdump-rtl-csa
  @opindex fdump-rtl-csa
  Dump after combining stack adjustments.
  
--- 5326,5336 ----
  @option{-fdump-rtl-ce3} enable dumping after the three
  if conversion passes.
  
! @item -fdump-rtl-cprop_hardreg
  @opindex fdump-rtl-cprop_hardreg
  Dump after hard register copy propagation.
  
! @item -fdump-rtl-csa
  @opindex fdump-rtl-csa
  Dump after combining stack adjustments.
  
***************
*** 5341,5351 ****
  @option{-fdump-rtl-cse1} and @option{-fdump-rtl-cse2} enable dumping after
  the two common sub-expression elimination passes.
  
! @itemx -fdump-rtl-dce
  @opindex fdump-rtl-dce
  Dump after the standalone dead code elimination passes.
  
! @itemx -fdump-rtl-dbr
  @opindex fdump-rtl-dbr
  Dump after delayed branch scheduling.
  
--- 5341,5351 ----
  @option{-fdump-rtl-cse1} and @option{-fdump-rtl-cse2} enable dumping after
  the two common sub-expression elimination passes.
  
! @item -fdump-rtl-dce
  @opindex fdump-rtl-dce
  Dump after the standalone dead code elimination passes.
  
! @item -fdump-rtl-dbr
  @opindex fdump-rtl-dbr
  Dump after delayed branch scheduling.
  
***************
*** 5390,5396 ****
  @opindex fdump-rtl-initvals
  Dump after the computation of the initial value sets.
  
! @itemx -fdump-rtl-into_cfglayout
  @opindex fdump-rtl-into_cfglayout
  Dump after converting to cfglayout mode.
  
--- 5390,5396 ----
  @opindex fdump-rtl-initvals
  Dump after the computation of the initial value sets.
  
! @item -fdump-rtl-into_cfglayout
  @opindex fdump-rtl-into_cfglayout
  Dump after converting to cfglayout mode.
  
***************
*** 5420,5426 ****
  @opindex fdump-rtl-rnreg
  Dump after register renumbering.
  
! @itemx -fdump-rtl-outof_cfglayout
  @opindex fdump-rtl-outof_cfglayout
  Dump after converting from cfglayout mode.
  
--- 5420,5426 ----
  @opindex fdump-rtl-rnreg
  Dump after register renumbering.
  
! @item -fdump-rtl-outof_cfglayout
  @opindex fdump-rtl-outof_cfglayout
  Dump after converting from cfglayout mode.
  
***************
*** 5432,5438 ****
  @opindex fdump-rtl-postreload
  Dump after post-reload optimizations.
  
! @itemx -fdump-rtl-pro_and_epilogue
  @opindex fdump-rtl-pro_and_epilogue
  Dump after generating the function prologues and epilogues.
  
--- 5432,5438 ----
  @opindex fdump-rtl-postreload
  Dump after post-reload optimizations.
  
! @item -fdump-rtl-pro_and_epilogue
  @opindex fdump-rtl-pro_and_epilogue
  Dump after generating the function prologues and epilogues.
  
Only in gcc-4.7.1-seaos/gcc/doc: invoke.texi.orig
diff -crB gcc-4.7.1/gcc/doc/sourcebuild.texi gcc-4.7.1-seaos/gcc/doc/sourcebuild.texi
*** gcc-4.7.1/gcc/doc/sourcebuild.texi	Mon Jul 25 09:44:22 2011
--- gcc-4.7.1-seaos/gcc/doc/sourcebuild.texi	Fri Apr  5 18:38:38 2013
***************
*** 676,682 ****
  @code{lang_checks}.
  
  @table @code
! @itemx all.cross
  @itemx start.encap
  @itemx rest.encap
  FIXME: exactly what goes in each of these targets?
--- 676,682 ----
  @code{lang_checks}.
  
  @table @code
! @item all.cross
  @itemx start.encap
  @itemx rest.encap
  FIXME: exactly what goes in each of these targets?
diff -crB gcc-4.7.1/libgcc/config.host gcc-4.7.1-seaos/libgcc/config.host
*** gcc-4.7.1/libgcc/config.host	Mon Apr 30 10:39:57 2012
--- gcc-4.7.1-seaos/libgcc/config.host	Fri Apr  5 18:38:36 2013
***************
*** 543,548 ****
--- 543,552 ----
  	tmake_file="${tmake_file} i386/t-crtpc i386/t-crtfm i386/t-crtstuff t-dfprules"
  	md_unwind_header=i386/linux-unwind.h
  	;;
+ i[3-7]86-*-seaos*)
+ 	extra_parts="$extra_parts crtbegin.o crtbeginS.o crtbeginT.o crtend.o crtendS.o"
+ 	tmake_file="${tmake_file} i386/t-crtpc i386/t-crtstuff"
+ 	;;
  i[34567]86-*-kfreebsd*-gnu | i[34567]86-*-knetbsd*-gnu | i[34567]86-*-gnu* | i[34567]86-*-kopensolaris*-gnu)
  	extra_parts="$extra_parts crtprec32.o crtprec64.o crtprec80.o crtfastmath.o"
  	tmake_file="${tmake_file} i386/t-crtpc i386/t-crtfm i386/t-crtstuff t-dfprules"
Only in gcc-4.7.1-seaos/libstdc++-v3: autom4te.cache
diff -crB gcc-4.7.1/libstdc++-v3/configure gcc-4.7.1-seaos/libstdc++-v3/configure
*** gcc-4.7.1/libstdc++-v3/configure	Sat Apr 21 06:28:53 2012
--- gcc-4.7.1-seaos/libstdc++-v3/configure	Fri Apr  5 18:39:45 2013
***************
*** 3025,3032 ****
  target_alias=${target_alias-$host_alias}
  
  # Handy for debugging:
! { $as_echo "$as_me:${as_lineno-$LINENO}: $build / $host / $target / $host_alias / $target_alias" >&5
! $as_echo "$as_me: $build / $host / $target / $host_alias / $target_alias" >&6;}; sleep 5
  
  if test "$build" != "$host"; then
    # We are being configured with some form of cross compiler.
--- 3025,3031 ----
  target_alias=${target_alias-$host_alias}
  
  # Handy for debugging:
! #AC_MSG_NOTICE($build / $host / $target / $host_alias / $target_alias); sleep 5
  
  if test "$build" != "$host"; then
    # We are being configured with some form of cross compiler.
***************
*** 11498,11504 ****
    lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
    lt_status=$lt_dlunknown
    cat > conftest.$ac_ext <<_LT_EOF
! #line 11501 "configure"
  #include "confdefs.h"
  
  #if HAVE_DLFCN_H
--- 11497,11503 ----
    lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
    lt_status=$lt_dlunknown
    cat > conftest.$ac_ext <<_LT_EOF
! #line 11500 "configure"
  #include "confdefs.h"
  
  #if HAVE_DLFCN_H
***************
*** 11604,11610 ****
    lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
    lt_status=$lt_dlunknown
    cat > conftest.$ac_ext <<_LT_EOF
! #line 11607 "configure"
  #include "confdefs.h"
  
  #if HAVE_DLFCN_H
--- 11603,11609 ----
    lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
    lt_status=$lt_dlunknown
    cat > conftest.$ac_ext <<_LT_EOF
! #line 11606 "configure"
  #include "confdefs.h"
  
  #if HAVE_DLFCN_H
***************
*** 14962,14968 ****
      #
      # Fake what AC_TRY_COMPILE does.  XXX Look at redoing this new-style.
      cat > conftest.$ac_ext << EOF
! #line 14965 "configure"
  struct S { ~S(); };
  void bar();
  void foo()
--- 14961,14967 ----
      #
      # Fake what AC_TRY_COMPILE does.  XXX Look at redoing this new-style.
      cat > conftest.$ac_ext << EOF
! #line 14964 "configure"
  struct S { ~S(); };
  void bar();
  void foo()
***************
*** 15297,15303 ****
    # Fake what AC_TRY_COMPILE does.
  
      cat > conftest.$ac_ext << EOF
! #line 15300 "configure"
  int main()
  {
    typedef bool atomic_type;
--- 15296,15302 ----
    # Fake what AC_TRY_COMPILE does.
  
      cat > conftest.$ac_ext << EOF
! #line 15299 "configure"
  int main()
  {
    typedef bool atomic_type;
***************
*** 15332,15338 ****
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15335 "configure"
  int main()
  {
    typedef short atomic_type;
--- 15331,15337 ----
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15334 "configure"
  int main()
  {
    typedef short atomic_type;
***************
*** 15367,15373 ****
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15370 "configure"
  int main()
  {
    // NB: _Atomic_word not necessarily int.
--- 15366,15372 ----
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15369 "configure"
  int main()
  {
    // NB: _Atomic_word not necessarily int.
***************
*** 15403,15409 ****
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15406 "configure"
  int main()
  {
    typedef long long atomic_type;
--- 15402,15408 ----
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15405 "configure"
  int main()
  {
    typedef long long atomic_type;
***************
*** 15482,15488 ****
    # unnecessary for this test.
  
      cat > conftest.$ac_ext << EOF
! #line 15485 "configure"
  int main()
  {
    _Decimal32 d1;
--- 15481,15487 ----
    # unnecessary for this test.
  
      cat > conftest.$ac_ext << EOF
! #line 15484 "configure"
  int main()
  {
    _Decimal32 d1;
***************
*** 15524,15530 ****
    # unnecessary for this test.
  
      cat > conftest.$ac_ext << EOF
! #line 15527 "configure"
  template<typename T1, typename T2>
    struct same
    { typedef T2 type; };
--- 15523,15529 ----
    # unnecessary for this test.
  
      cat > conftest.$ac_ext << EOF
! #line 15526 "configure"
  template<typename T1, typename T2>
    struct same
    { typedef T2 type; };
***************
*** 15558,15564 ****
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15561 "configure"
  template<typename T1, typename T2>
    struct same
    { typedef T2 type; };
--- 15557,15563 ----
      rm -f conftest*
  
      cat > conftest.$ac_ext << EOF
! #line 15560 "configure"
  template<typename T1, typename T2>
    struct same
    { typedef T2 type; };
***************
*** 51718,51723 ****
--- 51717,51868 ----
    fi
  
      ;;
+   *-seaos*)
+     for ac_header in sys/types.h locale.h float.h
+ do :
+   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
+ ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
+ eval as_val=\$$as_ac_Header
+    if test "x$as_val" = x""yes; then :
+   cat >>confdefs.h <<_ACEOF
+ #define `$as_echo "HAVE_$ac_header" | $as_tr_cpp` 1
+ _ACEOF
+ 
+ fi
+ 
+ done
+ 
+     GLIBCXX_CHECK_BUILTIN_MATH_SUPPORT
+     GLIBCXX_CHECK_COMPLEX_MATH_SUPPORT
+ 
+   ac_test_CXXFLAGS="${CXXFLAGS+set}"
+   ac_save_CXXFLAGS="$CXXFLAGS"
+   CXXFLAGS='-fno-builtin -D_GNU_SOURCE'
+ 
+ 
+   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for strtold declaration" >&5
+ $as_echo_n "checking for strtold declaration... " >&6; }
+   if test x${glibcxx_cv_func_strtold_use+set} != xset; then
+     if test "${glibcxx_cv_func_strtold_use+set}" = set; then :
+   $as_echo_n "(cached) " >&6
+ else
+ 
+ 
+       ac_ext=cpp
+ ac_cpp='$CXXCPP $CPPFLAGS'
+ ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
+ 
+       cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+ /* end confdefs.h.  */
+ #include <stdlib.h>
+ int
+ main ()
+ {
+  strtold(0, 0);
+   ;
+   return 0;
+ }
+ _ACEOF
+ if ac_fn_cxx_try_compile "$LINENO"; then :
+   glibcxx_cv_func_strtold_use=yes
+ else
+   glibcxx_cv_func_strtold_use=no
+ fi
+ rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+       ac_ext=c
+ ac_cpp='$CPP $CPPFLAGS'
+ ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ ac_compiler_gnu=$ac_cv_c_compiler_gnu
+ 
+ 
+ fi
+ 
+   fi
+   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $glibcxx_cv_func_strtold_use" >&5
+ $as_echo "$glibcxx_cv_func_strtold_use" >&6; }
+   if test x$glibcxx_cv_func_strtold_use = x"yes"; then
+     for ac_func in strtold
+ do :
+   ac_fn_c_check_func "$LINENO" "strtold" "ac_cv_func_strtold"
+ if test "x$ac_cv_func_strtold" = x""yes; then :
+   cat >>confdefs.h <<_ACEOF
+ #define HAVE_STRTOLD 1
+ _ACEOF
+ 
+ fi
+ done
+ 
+   fi
+ 
+ 
+ 
+ 
+   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for strtof declaration" >&5
+ $as_echo_n "checking for strtof declaration... " >&6; }
+   if test x${glibcxx_cv_func_strtof_use+set} != xset; then
+     if test "${glibcxx_cv_func_strtof_use+set}" = set; then :
+   $as_echo_n "(cached) " >&6
+ else
+ 
+ 
+       ac_ext=cpp
+ ac_cpp='$CXXCPP $CPPFLAGS'
+ ac_compile='$CXX -c $CXXFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ ac_link='$CXX -o conftest$ac_exeext $CXXFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ ac_compiler_gnu=$ac_cv_cxx_compiler_gnu
+ 
+       cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+ /* end confdefs.h.  */
+ #include <stdlib.h>
+ int
+ main ()
+ {
+  strtof(0, 0);
+   ;
+   return 0;
+ }
+ _ACEOF
+ if ac_fn_cxx_try_compile "$LINENO"; then :
+   glibcxx_cv_func_strtof_use=yes
+ else
+   glibcxx_cv_func_strtof_use=no
+ fi
+ rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
+       ac_ext=c
+ ac_cpp='$CPP $CPPFLAGS'
+ ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
+ ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
+ ac_compiler_gnu=$ac_cv_c_compiler_gnu
+ 
+ 
+ fi
+ 
+   fi
+   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $glibcxx_cv_func_strtof_use" >&5
+ $as_echo "$glibcxx_cv_func_strtof_use" >&6; }
+   if test x$glibcxx_cv_func_strtof_use = x"yes"; then
+     for ac_func in strtof
+ do :
+   ac_fn_c_check_func "$LINENO" "strtof" "ac_cv_func_strtof"
+ if test "x$ac_cv_func_strtof" = x""yes; then :
+   cat >>confdefs.h <<_ACEOF
+ #define HAVE_STRTOF 1
+ _ACEOF
+ 
+ fi
+ done
+ 
+   fi
+ 
+ 
+ 
+ 
+   CXXFLAGS="$ac_save_CXXFLAGS"
+ 
+     ;;
    *-mingw32*)
  
    # If we're not using GNU ld, then there's no point in even trying these
diff -crB gcc-4.7.1/libstdc++-v3/crossconfig.m4 gcc-4.7.1-seaos/libstdc++-v3/crossconfig.m4
*** gcc-4.7.1/libstdc++-v3/crossconfig.m4	Mon Jul 18 04:04:18 2011
--- gcc-4.7.1-seaos/libstdc++-v3/crossconfig.m4	Fri Apr  5 18:38:37 2013
***************
*** 150,155 ****
--- 150,161 ----
      GCC_CHECK_TLS
      AM_ICONV
      ;;
+   *-seaos*)
+     AC_CHECK_HEADERS([sys/types.h locale.h float.h])
+     GLIBCXX_CHECK_BUILTIN_MATH_SUPPORT
+     GLIBCXX_CHECK_COMPLEX_MATH_SUPPORT
+     GLIBCXX_CHECK_STDLIB_SUPPORT
+     ;;
    *-mingw32*)
      GLIBCXX_CHECK_LINKER_FEATURES
      GLIBCXX_CHECK_MATH_SUPPORT
