diff -crB e2fsprogs-1.42.2/config/config.sub e2fsprogs-seaos-1.42.2/config/config.sub
*** e2fsprogs-1.42.2/config/config.sub	2009-08-12 18:39:57.000000000 -0700
--- e2fsprogs-seaos-1.42.2/config/config.sub	2012-04-25 12:41:47.528565864 -0700
***************
*** 1264,1270 ****
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
  	      | -kopensolaris* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
--- 1264,1270 ----
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -solaris* | -sym* \
  	      | -kopensolaris* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* | -seaos* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -crB e2fsprogs-1.42.2/e2fsck/logfile.c e2fsprogs-seaos-1.42.2/e2fsck/logfile.c
*** e2fsprogs-1.42.2/e2fsck/logfile.c	2012-03-22 21:01:58.000000000 -0700
--- e2fsprogs-seaos-1.42.2/e2fsck/logfile.c	2012-04-25 12:43:03.171563138 -0700
***************
*** 122,136 ****
  			tm->tm_sec);
  		break;
  	case 'u':
! #ifdef TEST_PROGRAM
! 		strcpy(buf, "tytso");
  		break;
- #else
- 		getpwuid_r(getuid(), &pw_struct, buf, sizeof(buf), &pw);
- 		if (pw)
- 			append_string(s, pw->pw_name, 0);
- 		return;
- #endif
  	case 'U':
  		*flags |= FLAG_UTC;
  		return;
--- 122,129 ----
  			tm->tm_sec);
  		break;
  	case 'u':
! 		strcpy(buf, "god");
  		break;
  	case 'U':
  		*flags |= FLAG_UTC;
  		return;
diff -crB e2fsprogs-1.42.2/lib/ext2fs/bmap64.h e2fsprogs-seaos-1.42.2/lib/ext2fs/bmap64.h
*** e2fsprogs-1.42.2/lib/ext2fs/bmap64.h	2012-03-25 15:09:55.000000000 -0700
--- e2fsprogs-seaos-1.42.2/lib/ext2fs/bmap64.h	2012-04-25 12:42:11.245565022 -0700
***************
*** 8,14 ****
   * License.
   * %End-Header%
   */
! 
  struct ext2_bmap_statistics {
  	int		type;
  	struct timeval	created;
--- 8,14 ----
   * License.
   * %End-Header%
   */
! #include <sys/time.h>
  struct ext2_bmap_statistics {
  	int		type;
  	struct timeval	created;
diff -crB e2fsprogs-1.42.2/lib/quota/quota.h e2fsprogs-seaos-1.42.2/lib/quota/quota.h
*** e2fsprogs-1.42.2/lib/quota/quota.h	2012-03-18 11:18:33.000000000 -0700
--- e2fsprogs-seaos-1.42.2/lib/quota/quota.h	2012-04-25 12:45:44.091557347 -0700
***************
*** 36,41 ****
--- 36,53 ----
  #include <errno.h>
  #include <sys/types.h>
  #include <ext2fs/ext2_types.h>
+ #define u_int64_t unsigned long long
+ #define u_int32_t unsigned int
+ #define u_int16_t unsigned short
+ #define u_int8_t unsigned char
+ 
+ #define int64_t signed long long
+ #define int32_t signed int
+ #define int16_t signed short
+ #define int8_t signed char
+ 
+ #define intptr_t int*
+ #define u_intptr_t unsigned*
  
  #define __DQUOT_VERSION__	"dquot_6.5.2"
  
