diff -crB newlib-2.0.0/config.sub newlib-2.0.0-seaos/config.sub
*** newlib-2.0.0/config.sub	Fri Dec  7 16:02:34 2012
--- newlib-2.0.0-seaos/config.sub	Tue Mar 26 19:12:44 2013
***************
*** 1359,1365 ****
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
  	      | -sym* | -kopensolaris* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
--- 1359,1365 ----
  	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
  	      | -sym* | -kopensolaris* \
  	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
! 	      | -aos* | -aros* | -seaos* \
  	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
  	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
  	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff -crB newlib-2.0.0/newlib/configure.host newlib-2.0.0-seaos/newlib/configure.host
*** newlib-2.0.0/newlib/configure.host	Tue Dec  4 13:43:20 2012
--- newlib-2.0.0-seaos/newlib/configure.host	Tue Mar 26 19:13:52 2013
***************
*** 450,459 ****
  	##newlib_cflags="${newlib_cflags} -I`newlib-flags --kernel-dir`/include"
  	##newlib_cflags="${newlib_cflags} -idirafter ${gcc_dir}include"
  	;;
! 
    m68hc11-*-*|m6811-*-*|m6812-*-*|m68hc12-*-*)
  	;;
! 
    m68k-sun-sunos*)
  	unix_dir=unix
  	;;
--- 450,469 ----
  	##newlib_cflags="${newlib_cflags} -I`newlib-flags --kernel-dir`/include"
  	##newlib_cflags="${newlib_cflags} -idirafter ${gcc_dir}include"
  	;;
!   i[3-7]86-*-seaos*)
! 	sys_dir=seaos
! 	posix_dir=posix
! 	unix_dir=unix
! 	newlib_cflags="${newlib_cflags} -D_NO_EXECVE -DHAVE_GETOPT -D_I386MACH_ALLOW_HW_INTERRUPTS -DSIGNAL_PROVIDED -DHAVE_FCNT -DHAVE_OPENDIR -DHAVE_CLOSEDIR -DHAVE_READDIR -D_NO_GETCWD"
! 	;;
    m68hc11-*-*|m6811-*-*|m6812-*-*|m68hc12-*-*)
  	;;
!   x86_64-*-seaos*)
! 	sys_dir=seaos
! 	posix_dir=posix
! 	unix_dir=unix
! 	newlib_cflags="${newlib_cflags} -D_NO_EXECVE -DHAVE_GETOPT -D_I386MACH_ALLOW_HW_INTERRUPTS -DSIGNAL_PROVIDED -DHAVE_FCNT -DHAVE_OPENDIR -DHAVE_CLOSEDIR -DHAVE_READDIR -D_NO_GETCWD"
! 	;;
    m68k-sun-sunos*)
  	unix_dir=unix
  	;;
diff -crB newlib-2.0.0/newlib/libc/stdio/local.h newlib-2.0.0-seaos/newlib/libc/stdio/local.h
*** newlib-2.0.0/newlib/libc/stdio/local.h	Thu May 31 00:33:59 2012
--- newlib-2.0.0-seaos/newlib/libc/stdio/local.h	Tue Mar 26 19:12:44 2013
***************
*** 109,115 ****
  	}
  
  #endif /* !__SINGLE_THREAD__ && _POSIX_THREADS */
! 
  extern u_char *_EXFUN(__sccl, (char *, u_char *fmt));
  extern int    _EXFUN(__svfscanf_r,(struct _reent *,FILE *, _CONST char *,va_list));
  extern int    _EXFUN(__ssvfscanf_r,(struct _reent *,FILE *, _CONST char *,va_list));
--- 109,115 ----
  	}
  
  #endif /* !__SINGLE_THREAD__ && _POSIX_THREADS */
! #define u_char unsigned char
  extern u_char *_EXFUN(__sccl, (char *, u_char *fmt));
  extern int    _EXFUN(__svfscanf_r,(struct _reent *,FILE *, _CONST char *,va_list));
  extern int    _EXFUN(__ssvfscanf_r,(struct _reent *,FILE *, _CONST char *,va_list));
diff -crB newlib-2.0.0/newlib/libc/stdio/vfprintf.c newlib-2.0.0-seaos/newlib/libc/stdio/vfprintf.c
*** newlib-2.0.0/newlib/libc/stdio/vfprintf.c	Tue Oct 16 12:03:01 2012
--- newlib-2.0.0-seaos/newlib/libc/stdio/vfprintf.c	Tue Mar 26 19:12:44 2013
***************
*** 336,342 ****
   *
   * Make sure to avoid inlining.
   */
! _NOINLINE_STATIC int
  _DEFUN(__sbprintf, (rptr, fp, fmt, ap),
         struct _reent *rptr _AND
         register FILE *fp   _AND
--- 336,343 ----
   *
   * Make sure to avoid inlining.
   */
! 
! __attribute__ ((__noinline__)) static int
  _DEFUN(__sbprintf, (rptr, fp, fmt, ap),
         struct _reent *rptr _AND
         register FILE *fp   _AND
diff -crB newlib-2.0.0/newlib/libc/stdlib/mbstowcs.c newlib-2.0.0-seaos/newlib/libc/stdlib/mbstowcs.c
*** newlib-2.0.0/newlib/libc/stdlib/mbstowcs.c	Fri Apr 23 14:44:22 2004
--- newlib-2.0.0-seaos/newlib/libc/stdlib/mbstowcs.c	Tue Mar 26 19:12:44 2013
***************
*** 59,64 ****
--- 59,65 ----
          const char *s _AND
          size_t n)
  {
+   if (pwcs == 0) return n;
  #ifdef _MB_CAPABLE
    mbstate_t state;
    state.__count = 0;
diff -crB newlib-2.0.0/newlib/libc/sys/configure.in newlib-2.0.0-seaos/newlib/libc/sys/configure.in
*** newlib-2.0.0/newlib/libc/sys/configure.in	Tue Feb 21 14:34:31 2012
--- newlib-2.0.0-seaos/newlib/libc/sys/configure.in	Tue Mar 26 19:12:44 2013
***************
*** 36,41 ****
--- 36,42 ----
  	netware) AC_CONFIG_SUBDIRS(netware) ;;
  	rdos) AC_CONFIG_SUBDIRS(rdos) ;;
  	rtems) AC_CONFIG_SUBDIRS(rtems) ;;
+ 	seaos) AC_CONFIG_SUBDIRS(seaos) ;;
  	sh) AC_CONFIG_SUBDIRS(sh) ;;
  	sparc64) AC_CONFIG_SUBDIRS(sparc64) ;;
  	sun4) AC_CONFIG_SUBDIRS(sun4) ;;
